src/java/org/apache/cassandra/service/accord/AccordJournal.java

```diff
@@ -643,81 +654,94 @@ private void runOnce()
-                                    if (withEpochFailure instanceof Timeout)
-                                    {
-                                        node.withEpoch(waitForEpoch, this);
-                                        return;
-                                    }
-                                    else
-                                        throw new RuntimeException(withEpochFailure);
+                                    node.withEpoch(waitForEpoch, this);
+                                    return;
                                 }
-                                runOnce();
+                                else
+                                    throw new RuntimeException(withEpochFailure);
                             }
-                        };
-                        node.withEpoch(waitForEpoch, withEpochCallback);
-                    }
+                            runOnce();
+                        }
+                    };
+                    node.withEpoch(waitForEpoch, withEpochCallback);
+                }
 
-                    // Next, process all delayed epochs
-                    for (int i = 0; i < waitForEpochs.size(); i++)
+                // Next, process all delayed epochs
+                for (int i = 0; i < waitForEpochs.size(); i++)
+                {
+                    long epoch = waitForEpochs.getLong(i);
+                    if (node.topology().hasEpoch(epoch))
                     {
-                        long epoch = waitForEpochs.getLong(i);
-                        if (node.topology().hasEpoch(epoch))
+                        List<RequestContext> requests = byEpoch.remove(epoch);
+                        assert requests != null : String.format("%s %s (%d)", byEpoch, waitForEpochs, epoch);
+                        for (RequestContext request : requests)
                         {
-                            List<RequestContext> requests = byEpoch.remove(epoch);
-                            assert requests != null : String.format("%s %s (%d)", byEpoch, waitForEpochs, epoch);
-                            for (RequestContext request : requests)
+                            try
                             {
-                                try
-                                {
-                                    request.process(node, endpointMapper);
-                                }
-                                catch (Throwable t)
-                                {
-                                    logger.error(String.format("Caught an exception while processing a delayed request %s", request), t);
-                                }
+                                request.process(node, endpointMapper);
+                            }
+                            catch (Throwable t)
+                            {
+                                logger.error(String.format("Caught an exception while processing a delayed request %s", request), t);
```

Comment:
```suggestion
                                logger.error("Caught an exception while processing a delayed request {}", request, t);
```

I don't know why I keep seeing people use `String.format` w/e there is an exception... slf4j supports this case just fine...